drivers/rilmodem/voicecall.c

```diff
@@ -404,6 +419,14 @@ static void dial(struct ofono_voicecall *vc,
 		g_free(cbd);
 		CALLBACK_WITH_FAILURE(cb, data);
 	}
+
+	/*
+	 * The modem may issue an UNSOL_RESPONSE_CALL_STATE_CHANGED before
+	 * respond to the dial request. As we can notice the new call only at
+	 * the first CLCC, we have to put the callback in the vd right now.
+	 */
+	vd->cb = cb;
+	vd->data = data;
```

Comment:
I tend to forget that even the function already call the callback with a failure, it's still running.